observability/micrometer/src/main/java/org/wildfly/extension/micrometer/prometheus/PrometheusRegistryDefinitionRegistrar.java

```diff
@@ -112,4 +106,35 @@ public ResourceServiceInstaller configure(OperationContext context, ModelNode mo
                 .asActive()
                 .build();
     }
+
+    private static class AddHandler implements UnaryOperator<OperationStepHandler> {
+        @Override
+        public OperationStepHandler apply(OperationStepHandler operationStepHandler) {
+            return new AbstractAddStepHandler() {
```

Comment:
This may result in unexpected behavior - I would wrap this with a OperationStepHandler instead of some implementation.

Reply:
What you actually want is to decorate the OperationStepHandler apply(...) parameter with your own logic.  You do not actually want any of the logic inherited by AbstractAddStepHandler, so do not extend this.

Reply:
@pferraro How does this look now?

Reply:
LGTM

Reply:
@pferraro Thank you, Paul!